internal/git/git.go

```diff
@@ -38,18 +38,18 @@ func (c *checkoutCmd) String() string {
 	}
 
 	if commit != "" {
-		checkoutCommand = fmt.Sprintf("git clone %s %s && cd %s && git checkout %s && cp -r %s/* /manifests",
```

Comment:
I wonder whether we should just be copying the configured manifests directory (and any nested directories?) instead of the root of the repository? We could also be filtering for files that have the .yml or .yaml file extension to help alleviate a bloated volume. Any obvious downsides of this approach?

Reply:
In my mind, this code is meant to unpack arbitrary bundle content without any idea what that bundle content is. So I'm thinking that none of the source implementations should know anything about a `manifests` directory (or anything else).

Consider a helm chart provisioner. It should be able to re-use this git source implementation to unpack a chart directory, which doesn't contain a `manifests` directory and has a bunch of other important files.

If the entire repo is meant to be a bundle, then it makes sense to me that we'd copy the root of the repo. If a subdirectory is meant to be the bundle (e.g. `<repoDir>/bundle`), then the user would specify `bundle` as the repo directory, and we'd copy the full contents of that directory.

Reply:
Thanks for the explanation - I think that sounds reasonable to me. One thing to note is whether we'll continue to see the unpacker component continue to be agnostic, in addition to acting as the entrypoint for sourcing Bundle contents from various source types. 

We've thrown around the idea on an in-process g/go-containerregistry implementation for image-based source types, but I think it's entirely possible we'll play with a git Go client to handle the checking out process (e.g. tag/branch/etc. concrete implementations), and if that's the case, it feels like the unpacking component doesn't add a ton of value in the context of the overall sourcing process.

I'm fine with pushing this through though as it seems like a short-term win while we continue to iterate on the underlying sourcing implementation, and removing potentially bloated directories like .git/ works with me.

Reply:
> it feels like the unpacking component doesn't add a ton of value in the context of the overall sourcing process.

The way I've been thinking about this is that the `Bundle` source API is something we'll want all (or at least almost all) provisioners to implement, so I think we should build our source implementations to be agnostic to bundle types to decrease the effort required to build new provisioners.  If every provisioner has to write custom code to implement generic "pull a directory out of this artifact" code, that seems like it will result in every provisioner behaving slightly differently for really no reason other than it was implemented twice.

Another benefit of agnostic source unpackers is that we can keep tweaking/improving our plain+v0 spec without changing this implementation. And we'll be able to reuse it to unpack registry+v1 bundles.

Reply:
And I agree that `go-git` could be a better implementation for git sources, but I'm not sure I understand how that changes the value of an agnostic unpacker.

Reply:
Somewhat related: would it make sense to purge the vendor/ directory if it's present to help cut down on the size of the repository?